component/encryptor/encryptor.go

```diff
@@ -0,0 +1,368 @@
+package encryptor
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/binary"
+	"fmt"
+	"os"
+	"sync"
+	"sync/atomic"
+	"syscall"
+	"time"
+
+	"github.com/Azure/azure-storage-fuse/v2/common/config"
+	"github.com/Azure/azure-storage-fuse/v2/common/log"
+	"github.com/Azure/azure-storage-fuse/v2/internal"
+	"github.com/Azure/azure-storage-fuse/v2/internal/handlemap"
+)
+
+// Implements two mount approach with only one file for data and metadata.
+type Encryptor struct {
+	internal.BaseComponent
+	handle           *os.File
+	blockSize        uint64
+	mountPointCipher string
+	encryptionKey    []byte
+	lastChunkMeta    *LastChunkMeta
+}
+
+type LastChunkMeta struct {
+	sync.Mutex
+	farthestBlockSeen int64 // keep track of the farthest block that has been written to the file.
+	paddingLength     int64 // keep track of the padding length of the last block.
+}
+
+type encryptorOptions struct {
+	BlockSize          uint64 `config:"block-size-mb" yaml:"block-size-mb,omitempty"`
+	EncryptedMountPath string `config:"encrypted-mount-path" yaml:"encrypted-mount-path,omitempty"`
+	EncryptionKey      string `config:"encryption-key" yaml:"encryption-key,omitempty"`
+}
+
+const (
+	compName    = "encryptor"
+	AuthTagSize = 16
+	NonceSize   = 12
+	MetaSize    = 28 // Sum of AuthTagSize and NonceSize.
+)
+
+var _ internal.Component = &Encryptor{}
+
+func (e *Encryptor) Name() string {
+	return compName
+}
+
+func (e *Encryptor) SetName(name string) {
+	e.BaseComponent.SetName(name)
+}
+
+func (e *Encryptor) SetNextComponent(nc internal.Component) {
+	e.BaseComponent.SetNextComponent(nc)
+}
+
+func (e *Encryptor) Priority() internal.ComponentPriority {
+	return internal.EComponentPriority.LevelMid()
+}
+
+func (e *Encryptor) Configure(isParent bool) error {
+	log.Trace("Encryptor::Configure :  %s", e.Name())
+
+	conf := encryptorOptions{}
+	err := config.UnmarshalKey(e.Name(), &conf)
+	if err != nil {
+		log.Err("Encryptor::Configure : config error [invalid config attributes]")
+		return fmt.Errorf("error reading config for %s: %w", e.Name(), err)
+	}
+
+	key := os.Getenv("ENCRYPTION_KEY")
```

Comment:
>os.Getenv("ENCRYPTION_KEY") [](http://example.com/codeflow?start=8&length=27)

You shouldn't have to do this. Instead, you should bind the environment variable to the config via the BindEnv method [https://github.com/Azure/azure-storage-fuse/blob/main/common/config/config_parser.go#L200](https://github.com/Azure/azure-storage-fuse/blob/main/common/config/config_parser.go#L200)

See [https://github.com/Azure/azure-storage-fuse/blob/c4d638f7d07150899e09dc880d9dd0b4d61c3c32/component/azstorage/config.go#L198C6-L198C26](https://github.com/Azure/azure-storage-fuse/blob/c4d638f7d07150899e09dc880d9dd0b4d61c3c32/component/azstorage/config.go#L198C6-L198C26)